cpp/src/ice2slice/Gen.h

```diff
@@ -22,35 +22,38 @@ namespace Slice
     private:
         std::string _fileBase;
 
-        class OutputVisitor : public ParserVisitor
+        class OutputVisitor final : public ParserVisitor
```

Comment:
What about adding a comment for each visitor describing what it does?

OutputVisitor is not at all obvious to me, even after looking at its implementation.

Reply:
Add comments to the visitors